src-web/components/RecentRequestsDropdown.tsx

```diff
@@ -79,9 +80,12 @@ export function RecentRequestsDropdown({ className }: Pick<ButtonProps, 'classNa
     }
 
     return recentRequestItems.slice(0, 20);
-  }, [activeWorkspace, recentRequestIds, requests, routes, activeEnvironment?.id, activeCookieJar?.id]);
+  }, [activeWorkspace, allRecentRequestIds, requests, routes, activeEnvironment?.id, activeCookieJar?.id]);
+
+  // Only show dropdown if there are valid recent requests
+  const hasRecentRequests = items.length > 0 && items[0]?.key !== 'no-recent-requests';
```

Comment:
To improve this, I would make the `useMemo` return the type `{ items: DropdownItem[]; hasRecentRequests: boolean }`.